tests/test_config.py

```diff
@@ -395,3 +396,116 @@ def test_get_and_verify_max_len(model_id, max_model_len, expected_max_len,
     else:
         actual_max_len = model_config.get_and_verify_max_len(max_model_len)
         assert actual_max_len == expected_max_len
+
+
+@pytest.mark.parametrize("s3_url", [
+    "s3://air-example-data/rayllm-ossci/facebook-opt-350m/",
+    "s3://air-example-data/rayllm-ossci/meta-Llama-3.2-1B-Instruct/",
+])
+def test_s3_url_model_tokenizer_paths(s3_url):
+    """Test that S3 URLs create deterministic local directories for model and
+    tokenizer."""
+
+    # Create first config with S3 URL
+    config1 = ModelConfig(model=s3_url, tokenizer=s3_url)
+
+    # Check that model and tokenizer point to existing directories
+    assert os.path.exists(
+        config1.model), f"Model directory does not exist: {config1.model}"
+    assert os.path.isdir(
+        config1.model), f"Model path is not a directory: {config1.model}"
+    assert os.path.exists(
+        config1.tokenizer
+    ), f"Tokenizer directory does not exist: {config1.tokenizer}"
+    assert os.path.isdir(
+        config1.tokenizer
+    ), f"Tokenizer path is not a directory: {config1.tokenizer}"
+
+    # Verify that the paths are different from the original S3 URL
+    assert config1.model != s3_url, (
+        "Model path should be converted to local directory")
+    assert config1.tokenizer != s3_url, (
+        "Tokenizer path should be converted to local directory")
+
+    # Verify that the paths are deterministic based on the hash
+    import hashlib
```

Comment:
Let's remove this test since it's implementation dependent. Checking for the determinism through creating a config twice and seeing if the resulting paths match is the better test, which you are already doing.